ibis/expr/types/core.py

```diff
@@ -602,7 +602,11 @@ def unbind(self) -> ir.Table:
 
     def as_table(self) -> ir.Table:
         """Convert an expression to a table."""
-        raise NotImplementedError(type(self))
+        raise NotImplementedError("This expression cannot be converted to a table")
+
+    def as_scalar(self) -> ir.Scalar:
+        """Convert an expression to a scalar."""
+        raise NotImplementedError("This expression cannot be converted to a scalar")
```

Comment:
Similarly with `as_table`, showing the specific thing that cannot be converted into a scalar will help both users and developers understand where in their expression things are failing.

Reply:
Updated.